stdlib/test/tempfile/test_tempfile.mojo

```diff
@@ -0,0 +1,166 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+# RUN: %mojo %s
+
+import os
+from os.path import exists
+from pathlib import Path
+from testing import assert_true, assert_false, assert_equal
+from tempfile import gettempdir, mkdtemp
+
+
+fn test_mkdtemp() raises:
+    var dir_name: String
+
+    dir_name = mkdtemp()
+    assert_true(exists(dir_name), "Failed to create temporary directory")
+    os.rmdir(dir_name)
+    assert_false(exists(dir_name), "Failed to delete temporary directory")
+
+    dir_name = mkdtemp(prefix="my_prefix", suffix="my_suffix")
+    assert_true(exists(dir_name), "Failed to create temporary directory")
+    var name = dir_name.split("/")[-1]
+    assert_true(name.startswith("my_prefix"))
+    assert_true(name.endswith("my_suffix"))
+
+    os.rmdir(dir_name)
+    assert_false(exists(dir_name), "Failed to delete temporary directory")
+
+    dir_name = mkdtemp(dir=Path().__fspath__())
+    assert_true(exists(dir_name), "Failed to create temporary directory")
+    assert_true(
+        exists(Path() / dir_name.split("/")[-1]),
+        "Expected directory to be created in cwd",
+    )
+    os.rmdir(dir_name)
+    assert_false(exists(dir_name), "Failed to delete temporary directory")
+
+
+struct TempEnvWithCleanup:
+    var vars_to_set: Dict[String, String]
+    var _vars_back: Dict[String, String]
+    var clean_up_function: fn () raises -> None
+    """Function called after the context manager exits if an error occurs."""
+
+    fn __init__(
+        inout self,
+        vars_to_set: Dict[String, String],
+        clean_up_function: fn () raises -> None,
+    ):
```

Comment:
non-blocking: This is fine for now, since it's just a local testing utility. If we move this to `os` or `testing`, we should add an overload that takes variadic keyword arguments. We should also consider, whether `clean_up_function` should be a parameter.